### PR TITLE
The initial version of the Topic Modeling modules (v0.1.0)

### PR DESCRIPTION
**Pull request by keamanansiber**
The initial version of the Topic Modeling modules (v0.1.0), contributed by Hatma Suryotrisongko, consists of:
- maryam/core/util/iris/topic.py
- maryam/modules/iris/topicmodeling.py

TESTED USING =
topicmodeling -i mixed.json -t json -m all-distilroberta-v1
topicmodeling -i mixed.json -t json -s -v -m all-distilroberta-v1
topicmodeling -i mixed.json -t json -s -m all-distilroberta-v1
topicmodeling -i mixed.json -t json -v -m all-distilroberta-v1
topicmodeling -i testdataset.csv -t csv -m all-mpnet-base-v2
topicmodeling -i testdataset.csv -t csv -s -v -m all-mpnet-base-v2
topicmodeling -i testdataset.csv -t csv -s -m all-mpnet-base-v2
topicmodeling -i testdataset.csv -t csv -v -m all-mpnet-base-v2

Note: download the dataset for testing from https://github.com/keamanansiber/Maryam/tree/master/notebooks

### CODE REVIEW

**Comment by saeeddhqan** [comment]
I'll review the pr when there is no change.

**Comment by k4u5h1k** [comment]
Another error I got was this
image
This is because we have a script named `plotly.py` in the iris directory and python seems to think this is the plotly module. Please go ahead and rename it.

**Comment by keamanansiber** [comment]
I have followed your suggestion, by renaming the plotly.py

In addition, I also added the --output and --api functionality. 
Tested by using =
topicmodeling -i mixed.json -t json -m all-distilroberta-v1  --output
report json testreport iris/topicmodeling
topicmodeling -i mixed.json -t json -m all-distilroberta-v1  --api

File: maryam/modules/iris/topicmodeling.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		('keyword', None, False, 'Search keyword: ', '-k', 'store', str),
+		('showcharts', None, False, 'Show charts?', '-s', 'store_true', bool),
+		('verbose', None, False, 'Verbose output?', '-v', 'store_true', bool),
+		('pretrained_model', None, True, 'model for embedding', '-m', 'store', str),
+	),
+	'examples': ('topicmodeling -i mixed.json -t json -k music -s True -v False -m all-mpnet-base-v2')
+}
+
+
+def module_api(self):
```
Comment (saeeddhqan) [inline maryam/modules/iris/topicmodeling.py:39]: show the output in the module_run function. In this function, just prepare the output.

File: maryam/modules/iris/topicmodeling.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+	run = self.topic(self.options['inputfile'], self.options['filetype'], self.options['keyword'], self.options['showcharts'], self.options['verbose'])
+	run.run_sklearn_cluster_kmeans(self.options['pretrained_model'], self.options['showcharts'], self.options['verbose'])
+
+	results = run.run_topic_modeling_bertopic(self.options['pretrained_model'], self.options['verbose'])
+	print("\n\nResults = \n")
+	print( results )
+
+	output = {'results': results.to_json(orient="records") }
+	print("\n\nOutput = \n")
```
Comment (saeeddhqan) [inline maryam/modules/iris/topicmodeling.py]: use self.output instead of print

File: maryam/core/util/iris/topic.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
+# core/util/iris/topic.py
+# Hatma Suryotrisongko
+
+import pandas as pd
```
Comment (saeeddhqan) [inline maryam/core/util/iris/topic.py:4]: imports are heavy and decrease the speed of the framework. So, try to import them in the class.
also, add them all to the requirements file.

**Pull request merged**